src/App.js

```diff
@@ -1,26 +1,127 @@
+import foods from './foods.json';
+import FoodBox from './components/FoodBox';
+import AddFood from './components/AddFood';
+import DayFood from './components/DayFood';
+import 'bulma/css/bulma.css';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    allFood: foods,
+    filteredFood: foods,
+    todayFood: [],
+    showForm: false,
+    calories: 0
+  };
+
+  
+
+  handleSearch = (event) => {
+    const search = event.target.value;
+    const { allFood } = this.state;
+
+    let filteredFood = allFood.filter((food) => {
+      return food.name.toLowerCase().includes(search.toLowerCase());
+    });
+
+    this.setState({
+      filteredFood: filteredFood,
+    });
+  };
+
+  handleFormShow = () => {
+    this.setState({
+      showForm: true,
+    });
+  };
+
+  handleAdd = (event) => {
+    event.preventDefault();
+    let food = event.target;
+    const { allFood, filteredFood } = this.state;
+    let newFood = {
+      name: food.name.value,
+      image: food.image.value,
+      calories: food.calories.value,
+      quantity: 0,
+    };
+
+    this.setState({
+      allFood: [newFood, ...allFood],
+      filteredFood: [newFood, ...filteredFood],
+      showForm: false,
+    });
+  };
+
+  handleAddToDay = (n, q, c) => {
+    const { todayFood } = this.state;
+    const dayFood = {
+      name: n,
+      quantity: Number(q),
+      calories: c,
+    };
+    console.log(todayFood);
+    for (let food in todayFood) {
+      if (todayFood[food].name === n) {
+        todayFood[food].quantity += Number(q);
+        Number(todayFood[food].quantity);
+        return this.setState({
+          todayFood: [...this.state.todayFood],
+        });
+      }
+    }
+    this.setState({
+      todayFood: [...this.state.todayFood, dayFood],
+    });
+  };
+
+  render() {
+    return (
+      <div className="container">
+        {this.state.showForm ? (
+          <AddFood onAdd={this.handleAdd} />
+        ) : (
+          <button onClick={this.handleFormShow} class="button is-warning">
+            Show
+          </button>
+        )}
+          {/* Seacrh */}
+        <input
+          onChange={this.handleSearch}
+          className="input is-primary"
+          type="text"
+          placeholder="Search"
+        />
+
+        <div className="collumns">
+          <div className="collumn">
+          {/* displaying all the foods */}
+            {this.state.filteredFood.map((aFood) => {
+              return (
+                <FoodBox
```

Comment:
remember to add a key to every item rendered in a loop in react. This can be the index coming from the parameter